### PR TITLE
supply DQM for `SiPixelCalSingleMuonTight`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

With the foreseen increase of isolated muon trigger rates (see https://its.cern.ch/jira/browse/CMSHLT-2465 ) it becomes possible to run the `SiPixelCalSingleMuonTight` at express, allowing for quasi-online efficiency monitoring.
This PR adds a stub of DQM monitor for that AlCa flavour to start producing basic cluster properties histograms that are used as ingredients of the hit efficiency measurement.

#### PR validation:

Run

```console
 runTheMatrix.py -l 1002.4 -t 4 -j 8 --iboes
```

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, but will need to be backported

### CODE REVIEW

**Comment by mmusich** [comment]
type trk, new-feature

**Comment by mmusich** [comment]
test parameters:
   * workflow = 1002.4

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by emanueleusai** [comment]
+1

File: DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackerCommon/interface/PixelBarrelName.h"
+#include "DataFormats/TrackerCommon/interface/PixelEndcapName.h"
```
Comment (perrotta) [inline DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc:31]: Needed?

File: DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHit.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
```
Comment (perrotta) [inline DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc:37]:
```suggestion
```

File: DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/TrackerNumberingBuilder/interface/GeometricDet.h"
```
Comment (perrotta) [inline DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc:44]: Needed?

File: DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+#include "DataFormats/TrackerCommon/interface/PixelBarrelName.h"
+#include "DataFormats/TrackerCommon/interface/PixelEndcapName.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHit.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
```
Comment (perrotta) [inline DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc:39]: Needed?

**Comment by mmusich** [comment]
@cmsbuild, please test

File: DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+    }
+
+    const PixelGeomDetUnit* pixdet = (const PixelGeomDetUnit*)trackerGeometry_->idToDetUnit(detId);
+    edmNew::DetSet<SiPixelCluster>::const_iterator itCluster = itClusterSet->begin();
+    for (; itCluster != itClusterSet->end(); ++itCluster) {
+      LocalPoint lp(itCluster->x(), itCluster->y(), 0.);
+      PixelClusterParameterEstimator::ReturnType params = pixelCPE_->getParameters(*itCluster, *pixdet);
+      lp = std::get<0>(params);
+
+      float D = sqrt((lp.x() - traj_lx) * (lp.x() - traj_lx) + (lp.y() - traj_ly) * (lp.y() - traj_ly));
```
[the remark this answers is not in the document]
Comment (mmusich) [reply DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc]: done.

File: DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc (hunk 1 above)
Comment (mmusich) [reply DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc]: no.

File: DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc (hunk 4 above)
Comment (mmusich) [reply DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc]: no

File: DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc (hunk 3 above)
Comment (mmusich) [reply DQMOffline/CalibTracker/plugins/SiPixelCalSingleMuonAnalyzer.cc]: no

**Comment by mmusich** [comment]
>  found errors in the following unit tests:
> ---> test test-das-selected-lumis had ERRORS

test failure is unrelated.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
@emanueleusai , the changes since [your previous signature](https://github.com/cms-sw/cmssw/pull/39382#issuecomment-1246767555) are rather minor, and do not affect the results, I keep that signature for good and merge this PR in the master, aiming at merging its backports by tomorrow.

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

**Pull request merged**